lib/KSelect/index.vue

```diff
@@ -60,123 +167,606 @@
+      floatingLabel: {
         type: Boolean,
-        default: false,
+        default: true,
+      },
+      noResultsText: {
+        type: String,
+        default: '',
+      },
+      keys: {
+        type: Object,
+        default() {
+          return config.data.UiSelect.keys;
+        },
       },
-      /**
-       * Whether invalid or not
-       */
       invalid: {
         type: Boolean,
         default: false,
       },
-      /**
-       * Text displayed if invalid
-       */
       invalidText: {
         type: String,
         default: null,
       },
-      /**
-       * Whether or not display as inline block
-       */
-      inline: {
+      disabled: {
         type: Boolean,
         default: false,
       },
-      floatingLabel: {
+      clearable: {
         type: Boolean,
-        default: true,
+        default: false,
       },
-      placeholder: {
+      clearText: {
         type: String,
-        default: null,
+        default: '',
       },
       /**
-       * Whether to turn into a clearable state
-       * when an option has been selected.
+       * Whether or not display as inline block
        */
-      clearable: {
+      inline: {
         type: Boolean,
         default: false,
       },
-      clearText: {
-        type: String,
-        default: '',
-      },
     },
+
     data() {
       return {
+        query: '',
+        isInsideModal: false,
+        isActive: false,
+        isTouched: false,
+        highlightedOption: null,
+        showDropdown: false,
+        initialValue: JSON.stringify(this.value),
+        quickMatchString: '',
+        quickMatchTimeout: null,
+        scrollableAncestor: null,
+        dropdownButtonBottom: 'auto',
+        maxDropdownHeight: 256,
         // workaround for Keen-ui not displaying floating labels for empty objects
         selection: Object.keys(this.value || {}).length === 0 ? '' : this.value,
       };
     },
+
     computed: {
       name() {
         return `k-select-${this._uid}`;
       },
+      classes() {
+        return [
+          `ui-select-type-basic`,
+          { 'is-active': this.isActive },
+          { 'is-invalid': this.invalid },
+          { 'is-touched': this.isTouched },
+          { 'is-disabled': this.disabled },
+          { 'has-label': this.hasLabel },
+          { 'has-floating-label': this.hasFloatingLabel },
+          { 'k-select-inline': this.inline },
+          { 'k-select-disabled': this.disabled },
+        ];
+      },
+
+      labelClasses() {
+        return {
+          'is-inline': this.hasFloatingLabel && this.isLabelInline,
+          'is-floating': this.hasFloatingLabel && !this.isLabelInline,
+        };
+      },
+
+      hasLabel() {
+        return Boolean(this.label) || Boolean(this.$slots.default);
+      },
+
+      hasFloatingLabel() {
+        return this.hasLabel && this.floatingLabel;
+      },
+
+      isLabelInline() {
+        return this.selection.length === 0 && !this.isActive;
```

Comment:
I will get back to you on this. I think there was a reason. maybe for the initial value. But I will test it again and let you know.

Reply:
Hi @ozer550, I remembered the reason for using `this.selection` here. This variable is currently available in the [Kselect/index.vue](https://github.com/learningequality/kolibri-design-system/blob/develop/lib/KSelect/index.vue#L155) file. This holds the currently selected dropdown option and passes it to the `value` prop of `KeenUiSelect` component. So, whenever the dropdown option is selected, we emit the event and then the value of `selection` gets updated. The latest `selection` value is passed to the `KeenUiSelect` component and it displays it on the frontend. 

Since we have merged the code of `KeenUiSelect` and `KSelect` components we don't have a parent component to provide the currently selected option as `value` prop. We can't change the prop value that's why I added the `selection` state and am using it to display the currently selected option. `selection` initially holds the value of `value` prop.

Let me know if you have any other ideas.